### PR TITLE
Update Authenticator Definition

### PR DESCRIPTION
**Pull request by nicksteele**
***
<a href="https://pr-preview.s3.amazonaws.com/nicksteele/webauthn/pull/1195.html" title="Last updated on May 3, 2019, 4:10 PM UTC (34f17c1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1195/8678b43...nicksteele:34f17c1.html" title="Last updated on May 3, 2019, 4:10 PM UTC (34f17c1)">Diff</a>

### CODE REVIEW

**Comment by emlun** [comment]
Is there an issue corresponding to this PR?

**Comment by nicksteele** [comment]
Mostly #1175 - I wanted to draw more attention to the fact that the authenticator could be defined in either software or hardware

File: index.bs [hunk 1]
```diff
@@ -549,15 +549,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
 : <dfn>Authenticator</dfn>
 : <dfn>[WAA]</dfn>
-:: A cryptographic entity used by a [=[WAC]=] to (i) generate a [=public key credential=] and register it with a [=[RP]=],
-    and (ii) [=authentication|authenticate=] by potentially [=user verification|verifying the user=], and then 
-    cryptographically signing and returning, in the form of an [=Authentication Assertion=], 
-    a challenge and other data presented by a [=[WRP]=] (in concert with the [=[WAC]=]).
+:: A cryptographic entity, existing in hardware or software, that is capable of being used by a [=[WAC]=] to 
+    [=registration|register=] a user with a given [=[RP]=] and [=Authentication Assertion|asserting ownership=] of the 
+    registered [=public key credential=] when requested by the [=[RP]=]. [=Authenticators=] can report information regarding
+    their [=authenticator types|type=] and security characteristics via [=attestation=] during [=registration=].
```
Comment (emlun) [inline index.bs]: Although better than previously, the first sentence still feels a bit dense. Suggestions:

- s/is capable of being/can be/
- s/and [=Authentication Assertion|asserting ownership=]/and later [=Authentication Assertion|assert ownership=]/
- Maybe we can also drop "by a [=[WAC]=]"?

Result:

>A cryptographic entity, existing in hardware or software, that can be used to [=registration|register=] a user with a given [=[RP]=] and later [=Authentication Assertion|assert ownership=] of the registered [=public key credential=] when requested by the [=[RP]=].

**Comment by nicksteele** [comment]
Thanks for the edits! I think this is much more concise 👍

File: index.bs [hunk 2]
```diff
@@ -549,15 +549,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+:: A cryptographic entity, existing in hardware or software, that can [=registration|register=] a user with a given [=[RP]=] 
+    and later [=Authentication Assertion|assert ownership=] of the registered [=public key credential=] when requested by the [=[RP]=]. 
+    [=Authenticators=] can report information regarding their [=authenticator types|type=] and security characteristics via [=attestation=] 
+    during [=registration=].
 
-    A [=[WAA]=] could be, for example, a dedicated USB or wireless device separate from the [=client device=],
-    a dedicated hardware subsystem integrated into the [=client device=],
-    or a software component of the [=client=].
-    [=Authenticators=] can report their type and security characteristics via [=attestation=].
+    A [=[WAA]=] could be, for example, a [=roaming authenticator|USB or wireless device=] that is attached to the [=client device=],
```
Comment (akshayku) [inline index.bs]: Roaming authenticators are not attached. They are separate and communicate via various transports and need not be repeated here. Directly link to the definition.  

Change it to:
  A [=[WAA]=] could be, for example, a [=roaming authenticator=], a dedicated hardware subsystem in …

File: index.bs [hunk 3]
```diff
@@ -549,15 +549,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    a hardware subsystem integrated into the [=client device=], or a software component of the [=client=] or [=client device=].
```
Comment (akshayku) [inline index.bs]: a "dedicated" hardware.

**Comment by nicksteele** [comment]
The current definition doesn't mention that authenticators can be grounded in software, which I think is something important to denote. I think there is confusion currently around what an authenticator _is_, and this is a helpful point to make.

File: index.bs [hunk 4]
```diff
@@ -549,15 +549,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
 : <dfn>Authenticator</dfn>
 : <dfn>[WAA]</dfn>
-:: A cryptographic entity used by a [=[WAC]=] to (i) generate a [=public key credential=] and register it with a [=[RP]=],
-    and (ii) [=authentication|authenticate=] by potentially [=user verification|verifying the user=], and then 
-    cryptographically signing and returning, in the form of an [=Authentication Assertion=], 
-    a challenge and other data presented by a [=[WRP]=] (in concert with the [=[WAC]=]).
+:: A cryptographic entity, existing in hardware or software, that can [=registration|register=] a user with a given [=[RP]=] 
+    and later [=Authentication Assertion|assert ownership=] of the registered [=public key credential=] when requested by the [=[RP]=].
```
Comment (equalsJeffH) [inline index.bs]: minor personal-view comment y'all can ignore:   IMHO we way {mis,over}use the term "owner". I would s/ownership/possession/

**Comment by agl** [comment]
> The current definition doesn't mention that authenticators can be grounded in software

Current wording includes &ldquo;or a software component of the [=client=]&rdquo;.

**Comment by selfissued** [comment]
I believe the wording that @agl quotes does the trick.

**Comment by nicksteele** [comment]
It isn't only "of the [=client=]" though, it could also be a component of the client device.

**Comment by nicksteele** [comment]
@akshayku can you approve that the changes from your comments have been added?

File: index.bs [hunk 5]
```diff
@@ -549,15 +549,13 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
 : <dfn>Authenticator</dfn>
 : <dfn>[WAA]</dfn>
-:: A cryptographic entity used by a [=[WAC]=] to (i) generate a [=public key credential=] and register it with a [=[RP]=],
-    and (ii) [=authentication|authenticate=] by potentially [=user verification|verifying the user=], and then 
-    cryptographically signing and returning, in the form of an [=Authentication Assertion=], 
-    a challenge and other data presented by a [=[WRP]=] (in concert with the [=[WAC]=]).
+:: A cryptographic entity, existing in hardware or software, that can [=registration|register=] a user with a given [=[RP]=] 
+    and later [=Authentication Assertion|assert possession=] of the registered [=public key credential=] when requested by the [=[RP]=].
```
Comment (equalsJeffH) [inline index.bs]: I don't think we want to loose the notion of user verification:
suggest:
...assert possession=] of the registered [=public key credential=], and optionally [=user verification|verifying the user=],  when requested by the [=[RP]=].

File: index.bs (hunk 4 above)
Comment (equalsJeffH) [reply index.bs]: this was changed, thx.

**Comment by nadalin** [comment]
@nicksteele Please see @equalsJeffH comments, and update so we can merge this

**Comment by nicksteele** [comment]
Changes have been made and are ready for review @equalsJeffH

**Pull request merged**